### DISCUSSION TITLE
Add ability to auto generate a password when creating a new user

### DISCUSSION THREAD

**Discussion by maartenvn** [directus/directus#3051] [Feature Requests]
Add ability to auto generate a password when creating a new user

**Comment by benhaynes** [comment]
Normally I would move this to discussions, but I think it's a really useful feature to have. It will definitely need to be in the App, perhaps as an option/feature of the `password` interface. @Nitwel thoughts on that?

For the API, I'm not sure if we'd want to have this as an option, or just require it to be sent in the request and you can use a "generate password" endpoint. I think we have this endpoint in our utils already, but I defer to @rijkvanzanten on this.

**Comment by benhaynes** [comment]
In addition to a regenerate/copy button, these are the options that I typically like having control over. We might opt for something far more simple... but I could imagine something like this being in a password interface dropdown.

![Screen Shot 2020-10-02 at 8 20 00 AM](https://user-images.githubusercontent.com/522079/94922457-3ff4c000-0488-11eb-8396-fc363e36fa8f.png)

**Comment by Nitwel** [comment]
It is a nice idea though your image @benhaynes is a bit too much for my personal preference.

I would make it either a single clickable icon that either generates a string directly or a menu that has `normal`, `medium` and `difficult` as options. (Names could vary)

**Comment by maartenvn** [comment]
This wouldn't require an API endpoint, right? It could just be generated by the app.
I would also choose for just a simple dropdown for generating the password. I would only use it to set an initial password that is secure. It is up to the user to change it if they find some characters to difficult to type :smile:

**Comment by Nitwel** [comment]
Yep, there is no vulnerability in generating this in the app. 👍

**Comment by benhaynes** [comment]
True, this can be App/Interface only. I agree that simple is better... but I don't think a single "Generate" button is customizable enough, unless we set that once in the interface options. Otherwise if we're going to have a dropdown with "strong/normal/etc", then I think it's just as easy to have something like this:

![Screen Shot 2020-10-02 at 9 15 50 AM](https://user-images.githubusercontent.com/522079/94927165-e7292580-048f-11eb-878d-5b3a17c5cd86.png)


Do ya'll think this is too complex?

**Comment by Nitwel** [comment]
This is looking amazing!
My concern here is that this feature is pretty edgecase / close to hitting that 80/20 Rule.
As @maartenvn said, he would only use it to generate a temporarly password for users.

**Comment by benhaynes** [comment]
Potentially 80/20... but @maartenvn is an admin, and this is the password field in general, which means it also is for all users too. We could have an interface option to toggle this off entirely... and could even disable certain things in that dropdown based on the project's security settings (that determine the min password strength). 

"Normal/Strong/Secure" are good intro options, but I think more people are comfortable with this sort of control... since password managers are more common these days. Also, this helps us be a bit more "security first" than other platforms that "dumb things down" a bit too much.  And since we're going to be positioned as a "data platform", I could imagine a lot of our users will be more technical.

Not trying to push this direction, just trying to give a more wholistic view of our potential use-cases. 😄 

@rijkvanzanten ?

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
